Source/utils/stdcompat/shared_ptr_array.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <cstddef>
+#include <memory>
+
+namespace devilution {
+
+// Apple Clang 12 has a buggy implementation that fails to compile `std::shared_ptr<T[]>(new T[size])`.
+#if __cplusplus >= 201611L && (!defined(__clang_major__) || __clang_major__ >= 13)

```

Comment:
Oh my god at this line.....

Reply:
Is this a constructive comment or should I just brush it off?